src/subsystems/localization.rs

```diff
@@ -0,0 +1,469 @@
+///
+/// Provides functionality to send Loco Positioning Protocol (LPP) packets
+/// to ultra-wide-band positioning nodes.
+pub struct LocoPositioning {
+    uplink: Sender<Packet>,
+}
+
+impl LocoPositioning {
+    /// Send Loco Positioning Protocol (LPP) packet to a specific destination
+    ///
+    /// # Arguments
+    /// * `dest_id` - Destination node ID
+    /// * `data` - LPP packet payload
+    pub async fn send_short_lpp_packet(&self, dest_id: u8, data: &[u8]) -> Result<()> {
+        let mut payload = Vec::with_capacity(2 + data.len());
+        payload.push(LPS_SHORT_LPP_PACKET);
+        payload.push(dest_id);
+        payload.extend_from_slice(data);
+
+        let pk = Packet::new(LOCALIZATION_PORT, GENERIC_CHANNEL, payload);
+        self.uplink.send_async(pk).await.map_err(|_| Error::Disconnected)?;
+        Ok(())
+    }
+}
+
+/// Lighthouse positioning system interface
+///
+/// Provides functionality to receive lighthouse sweep angle data and manage
+/// lighthouse base station configuration persistence.
+pub struct Lighthouse {
+    uplink: Sender<Packet>,
+    angle_stream_receiver: BroadcastReceiver<LighthouseAngleData>,
+    persist_receiver: BroadcastReceiver<bool>,
+}
+
+impl Lighthouse {
+    /// Get a stream of lighthouse angle measurements
+    ///
+    /// Returns a Stream that yields [LighthouseAngleData] whenever lighthouse
+    /// sweep angle data is received from the Crazyflie. This is typically used
+    /// for lighthouse base station calibration and geometry estimation.
+    ///
+    /// To enable the angle stream, set the parameter `locSrv.enLhAngleStream` to 1
+    /// on the Crazyflie.
+    ///
+    /// # Example
+    /// ```no_run
+    /// # use crazyflie_lib::Crazyflie;
+    /// # use futures::StreamExt;
+    /// # async fn example(crazyflie: &Crazyflie) -> Result<(), Box<dyn std::error::Error>> {
+    /// // Enable angle streaming
+    /// crazyflie.param.set("locSrv.enLhAngleStream", 1u8).await?;
+    ///
+    /// let mut angle_stream = crazyflie.localization.lighthouse.angle_stream().await;
+    /// while let Some(data) = angle_stream.next().await {
+    ///     println!("Base station {}: x={:?}, y={:?}",
+    ///         data.base_station, data.angles.x, data.angles.y);
+    /// }
+    /// # Ok(())
+    /// # }
+    /// ```
+    pub async fn angle_stream(&self) -> impl Stream<Item = LighthouseAngleData> {
+        self.angle_stream_receiver.clone()
+    }
+
+    /// Wait for lighthouse persistence confirmation
+    ///
+    /// After sending geometry or calibration data to be persisted (via
+    /// send_lh_persist_data_packet), this function waits for and returns
+    /// the confirmation from the Crazyflie.
+    ///
+    /// Returns `Ok(true)` if data was successfully persisted, `Ok(false)` if
+    /// persistence failed, or an error if no confirmation is received within
+    /// the timeout.
+    pub async fn wait_persist_confirmation(&self) -> Result<bool> {
```

Comment:
What about adding this to `send_lh_persist_data_packet` so the user doesn't need to bother with calling the wait function (and can't make the mistake of not doing it either)?

Reply:
Good idea

Reply:
Fixed in b909392c7b0cc760025024f486291093f48b066e